querygen/querygen.go

```diff
@@ -0,0 +1,96 @@
+package querygen
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+var (
+	UserIdExpr = `if(isnull('impersonatedUser.username'),` +
+		`'user.username','impersonatedUser.username')`
+	IncludeFieldsCmd = `fields ` +
+		`messageId, timestamp, type, userId, event_verb, event_subject, properties`
+	ExcludeFieldsCmd = `fields - _*`
+)
+
+func GenDedupEval(fields []string) string {
```

Comment:
I think [Doc-comments](https://go.dev/doc/comment)  (=GO docstrings) will improve readability of this package (applicable in multiple parts of the file)

Reply:
added a bunch of comments.